File: test/fixtures/go/j_post_form_f.go
package main

import (
	"bytes"
	"fmt"
	"io"
	"log"
	"mime/multipart"
	"net/http"
)

func main() {
	form := new(bytes.Buffer)
	writer := multipart.NewWriter(form)
	formField, err := writer.CreateFormField("d1")
	if err != nil {
		log.Fatal(err)
	}
	_, err = formField.Write([]byte("data1"))

	formField, err = writer.CreateFormField("d2")
	if err != nil {
		log.Fatal(err)
	}
	_, err = formField.Write([]byte("data"))

	writer.Close()

	client := &http.Client{}
	req, err := http.NewRequest("POST", "http://localhost:28139/post", form)
	if err != nil {
		log.Fatal(err)
	}
	req.Header.Set("Content-Type", writer.FormDataContentType())
	resp, err := client.Do(req)
	if err != nil {
		log.Fatal(err)
	}
	defer resp.Body.Close()
	bodyText, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("%s\n", bodyText)
}

File: test/fixtures/csharp/j_post_form_f.cs
using System.Net.Http;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "http://localhost:28139/post");

MultipartFormDataContent content = new MultipartFormDataContent();
content.Add(new StringContent("data1"), "d1");
content.Add(new StringContent("data"), "d2");
request.Content = content;

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();
